optuna/trial/_trial.py

```diff
@@ -561,6 +564,12 @@ def objective(trial):
                 ``n_warmup_steps`` as the warmup mechanism.
         """
 
+        if len(self.study.directions) > 1:
+            _logger.warning(
+                "When the problem is multi-objective optimization, this function does nothing."
+            )
```

Comment:
Users may not find the cause of this warning because the message does not contain method name:
```console
[W 2020-12-22 10:18:22,724] When the problem is multi-objective optimization, this function does nothing.
[W 2020-12-22 10:18:22,724] When the problem is multi-objective optimization, this function always returns `False`.
[I 2020-12-22 10:18:22,724] Trial 99 finished with values: [47.14745396812321, 14.493481802459023] and parameters: {'x': 2.9140067465104154, 'y': 1.8153314224467625}.
```

Reply:
I added the name of the method. PTAL.